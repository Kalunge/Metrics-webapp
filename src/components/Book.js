import React from 'react';
import PropTypes from 'prop-types';
import { FaGlobeAfrica } from 'react-icons/fa';
import { AiOutlineArrowRight } from 'react-icons/ai';
import { useHistory } from 'react-router-dom';
// import reading from '../img/reading.jpg';

const Book = ({ book }) => {
  const history = useHistory();
  const handleClick = (abbrev) => {
    history.push(`/${abbrev}`);
  };
  return (
    <div
      role="button"
      tabIndex={0}
      className="col-6 book-outline"
      onClick={() => handleClick(book.abbrev.en)}
      onKeyDown={() => handleClick(book.abbrev.en)}
    >
      <AiOutlineArrowRight className="fachevron" />
      <div className="book">
        <FaGlobeAfrica className="fa-icons" size={120} />
        <h4 className="text">{`${book.name}: ${book.chapters}`}</h4>
      </div>
    </div>
  );
};

Book.propTypes = {
  book: PropTypes.shape({
    name: PropTypes.string.isRequired,
    abbrev: PropTypes.shape({
      en: PropTypes.string,
    }),
    chapters: PropTypes.number.isRequired,
  }).isRequired,
};
export default Book;
