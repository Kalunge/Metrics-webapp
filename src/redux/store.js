import { createStore, applyMiddleware, combineReducers } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import booksReducer from './books/books';

const middleware = [thunk, logger];

const reducer = combineReducers({
  books: booksReducer,
});
const store = createStore(reducer, applyMiddleware(...middleware));

export default store;
