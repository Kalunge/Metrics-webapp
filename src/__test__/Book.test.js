import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Book from '../components/Book';
import store from '../redux/store';
import book from './bookMOck';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Book book={book} />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
