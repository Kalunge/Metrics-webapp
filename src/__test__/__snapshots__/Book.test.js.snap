// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="col-6 book-outline"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={0}
>
  <svg
    className="fachevron"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 1024 1024"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M869 487.8L491.2 159.9c-2.9-2.5-6.6-3.9-10.5-3.9h-88.5c-7.4 0-10.8 9.2-5.2 14l350.2 304H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h585.1L386.9 854c-5.6 4.9-2.2 14 5.2 14h91.5c1.9 0 3.8-.7 5.2-2L869 536.2a32.07 32.07 0 0 0 0-48.4z"
    />
  </svg>
  <div
    className="book"
  >
    <svg
      className="fa-icons"
      fill="currentColor"
      height={120}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 496 512"
      width={120}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M248 8C111.03 8 0 119.03 0 256s111.03 248 248 248 248-111.03 248-248S384.97 8 248 8zm160 215.5v6.93c0 5.87-3.32 11.24-8.57 13.86l-15.39 7.7a15.485 15.485 0 0 1-15.53-.97l-18.21-12.14a15.52 15.52 0 0 0-13.5-1.81l-2.65.88c-9.7 3.23-13.66 14.79-7.99 23.3l13.24 19.86c2.87 4.31 7.71 6.9 12.89 6.9h8.21c8.56 0 15.5 6.94 15.5 15.5v11.34c0 3.35-1.09 6.62-3.1 9.3l-18.74 24.98c-1.42 1.9-2.39 4.1-2.83 6.43l-4.3 22.83c-.62 3.29-2.29 6.29-4.76 8.56a159.608 159.608 0 0 0-25 29.16l-13.03 19.55a27.756 27.756 0 0 1-23.09 12.36c-10.51 0-20.12-5.94-24.82-15.34a78.902 78.902 0 0 1-8.33-35.29V367.5c0-8.56-6.94-15.5-15.5-15.5h-25.88c-14.49 0-28.38-5.76-38.63-16a54.659 54.659 0 0 1-16-38.63v-14.06c0-17.19 8.1-33.38 21.85-43.7l27.58-20.69a54.663 54.663 0 0 1 32.78-10.93h.89c8.48 0 16.85 1.97 24.43 5.77l14.72 7.36c3.68 1.84 7.93 2.14 11.83.84l47.31-15.77c6.33-2.11 10.6-8.03 10.6-14.7 0-8.56-6.94-15.5-15.5-15.5h-10.09c-4.11 0-8.05-1.63-10.96-4.54l-6.92-6.92a15.493 15.493 0 0 0-10.96-4.54H199.5c-8.56 0-15.5-6.94-15.5-15.5v-4.4c0-7.11 4.84-13.31 11.74-15.04l14.45-3.61c3.74-.94 7-3.23 9.14-6.44l8.08-12.11c2.87-4.31 7.71-6.9 12.89-6.9h24.21c8.56 0 15.5-6.94 15.5-15.5v-21.7C359.23 71.63 422.86 131.02 441.93 208H423.5c-8.56 0-15.5 6.94-15.5 15.5z"
      />
    </svg>
    <h4
      className="text"
    >
      myBook: 5
    </h4>
  </div>
</div>
`;
